src/Compilers/CSharp/Test/Semantic/Semantics/SpanStackSafetyTests.cs

```diff
@@ -20,31 +20,60 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
     public class SpanStackSafetyTests : CompilingTestBase
     {
         private static string spanSource = @"
-
 namespace System
 {
-    public struct Span<T>
+    public ref struct Span<T>
     {
         public ref T this[int i] => throw null;
         public override int GetHashCode() => 1;
     }
 
-    public struct ReadonlySpan<T>
+    public ref struct ReadonlySpan<T>
     {
         public ref readonly T this[int i] => throw null;
         public override int GetHashCode() => 2;
     }
+
+    public ref struct SpanLike<T>
+    {
+        public Span<T> field;
+    }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Struct)]
+    public sealed class IsByRefLikeAttribute: Attribute
+    {
+        public IsByRefLikeAttribute(){}
+    }
 }
+
 ";
         //PROTOTYPE(span): this will be updated when rules for defining span are implemented
         //                 most likely we would just pick the actual binary/corlib where
         //                 span lives.
         private static CSharpCompilation CreateCompilationWithMscorlibAndSpan(string text, CSharpCompilationOptions options = null)
         {
+            var reference = CreateCompilationWithMscorlib45(
+                spanSource,
+                references: new List<MetadataReference>() { MscorlibRef_v4_0_30316_17626, SystemCoreRef, CSharpRef },
+                options: TestOptions.ReleaseDll).EmitToImageReference();
+
+            var comp = CreateCompilationWithMscorlib45(
+                text, 
+                references: new List<MetadataReference>() { MscorlibRef_v4_0_30316_17626, SystemCoreRef, CSharpRef, reference}, 
+                options: options ?? TestOptions.ReleaseExe);
```

Comment:
Should these two calls use `CreateCompilation()` instead since the corlib is provided explicitly?